### ISSUE TITLE
conv_transpose3d precision issues causing flaky tests for `vmapjvp` tests

### ISSUE THREAD

**Issue by Chillee** [pytorch/functorch#531]
`test_vmapjvpall_nn_functional_conv_transpose3d_cuda_float32`
```
Traceback (most recent call last):
  File "/home/circleci/project/env/lib/python3.8/site-packages/torch/testing/_internal/common_utils.py", line 1745, in wrapper
    method(*args, **kwargs)
  File "/home/circleci/project/env/lib/python3.8/site-packages/torch/testing/_internal/common_utils.py", line 1745, in wrapper
    method(*args, **kwargs)
  File "/home/circleci/project/env/lib/python3.8/site-packages/torch/testing/_internal/common_device_type.py", line 376, in instantiated_test
    result = test(self, **param_kwargs)
  File "/home/circleci/project/env/lib/python3.8/site-packages/torch/testing/_internal/common_device_type.py", line 761, in test_wrapper
    return test(*args, **kwargs)
  File "test/test_ops.py", line 774, in test_vmapjvpall
    self.assertEqual(loop_out, batched_out, atol=1e-4, rtol=1e-4)
  File "/home/circleci/project/env/lib/python3.8/site-packages/torch/testing/_internal/common_utils.py", line 2112, in assertEqual
    assert_equal(
  File "/home/circleci/project/env/lib/python3.8/site-packages/torch/testing/_comparison.py", line 1064, in assert_equal
    raise error_metas[0].to_error()
AssertionError: Tensor-likes are not close!
AssertionError: Tensor-likes are not close!

Mismatched elements: 1 / 10976 (0.0%)
Greatest absolute difference: 0.000102996826171875 at index (3, 0, 1, 3, 4, 3) (up to 0.0001 allowed)
Greatest relative difference: 0.008538899430740038 at index (3, 0, 1, 3, 4, 3) (up to 0.0001 allowed)


The failure occurred for item [0]
```

**Comment by zou3519** [comment]
This is expected: convolution is non-deterministic. We probably want to relax the tolerance

**Comment by Chillee** [comment]
Specifically, that's why this commit failed: https://app.circleci.com/pipelines/github/pytorch/functorch/1721/workflows/80f5785b-173a-4c0a-8c79-dbdfaa8a2ea8/jobs/10280

**Comment by ezyang** [comment]
affected me too https://app.circleci.com/pipelines/github/pytorch/functorch/1834/workflows/61225e3f-d78e-4cc3-9feb-65789325a706/jobs/10947

**Comment by zou3519** [comment]
So there is not an easy way to modify the OpInfo from outside of PyTorch core! The mechanism for overriding the precision for OpInfos is:
1. someone has to add a precision override into the OpInfo
2. but the OpInfo lives inside PyTorch core
We can easily hack this to work (by grabbing the OpInfo, monkeypatching its contents, then sending it to our test) but that doesn't feel very good

**Comment by samdow** [comment]
> We can easily hack this to work (by grabbing the OpInfo, monkeypatching its contents, then sending it to our test) but that doesn't feel very good

Definitely something to think about long term if we're bumping things out of core but want them to use OpInfos. For now though we are already patching the OpInfo decorators for [skipOps](https://github.com/pytorch/functorch/blob/main/test/common_utils.py#L174) right? So we could build something similar for toleranceOverrides

**Comment by zou3519** [comment]
Should be fixed by #560
